anker/src/logic.rs

```diff
@@ -230,6 +234,7 @@ pub fn swap_rewards(
     amount: StLamports,
     anker: &Anker,
     accounts: &SellRewardsAccountsInfo,
+    minimum_ust_price: MicroUst,
```

Comment:
This name looks wrong, it’s called `minimum_ust_price`, so I assume the unit would be something different than UST, e.g. the UST price in stSOL. If the unit is UST, then it’s really the price of stSOL (in UST).

Second, it’s passed as `minimum_amount_out`, so it’s not the price, it’s the price. Either we need to set `minimum_amount_out` to `amount * price`, or instead of price we take `minimum_ust_out` as an argument.

**Edit**: So from the calling code, it looks like this is the minimum stSOL price in UST. So we should call it `minimum_st_sol_price` or maybe even `minimum_st_sol_price_in_ust`, and we should set `minimum_amount_out` below to `amount * minimum_st_sol_price` (it needs some juggling with rationals to get the scale right).

Reply:
I changed it to `minimum_ust_out` and renamed the function to `minimum_ust_swap_amount`.
I think it's better to do the rewards calculation inside the function